src/libumf.c

```diff
@@ -0,0 +1,53 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <stddef.h>
+
+#include "base_alloc_global.h"
+#include "memspace_internal.h"
+#include "provider_tracking.h"
+#include "topology.h"
+#include "utils_log.h"
+
+umf_memory_tracker_handle_t TRACKER = NULL;
+
+static unsigned long long umfRefCount = 0;
+
+// Increment the usage reference counter and initialize the global state of libumf
+// if the usage reference counter was equal to 0.
+// Returns 0 on success or -1 on failure.
+// It must be called just after dlopen() and it is not required in other scenarios.
```

Comment:
I think you don't have to duplicate the comments from the header file

Reply:
Done